include/boost/beast2/body_write_stream.hpp

```diff
@@ -0,0 +1,248 @@
+//
+// Copyright (c) 2025 Mungo Gill
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/cppalliance/beast2
+//
+
+#ifndef BOOST_BEAST2_BODY_WRITE_STREAM_HPP
+#define BOOST_BEAST2_BODY_WRITE_STREAM_HPP
+
+#include <boost/asio/async_result.hpp>
+#include <boost/http_proto/serializer.hpp>
+#include <boost/system/error_code.hpp>
+
+namespace boost {
+namespace beast2 {
+
+/** A body writer for HTTP/1 messages.
+
+    This type meets the requirements of asio's AsyncWriteStream, and is
+    constructed with a reference to an underlying AsyncWriteStream.
+
+    Any call to `async_write_some` initially triggers writes from the underlying
+    stream until all of the HTTP headers and at least one byte of the body have
+    been write and processed.  Thereafter, each subsequent call to
+    `async_write_some` processes at least one byte of the body, triggering, where
+    required, calls to the underlying stream's `async_write_some` method.  The
+    resulting body data is stored in the referenced MutableBufferSequence.
+
+    All processing depends on a beast2::parser object owned by the caller and
+    referenced in the construction of this object.
+
+    @see
+        @ref http_proto::parser.
+*/
+template<class AsyncWriteStream>
+class body_write_stream
+{
+
+public:
+    /** The type of the executor associated with the stream.
+
+        This will be the type of executor used to invoke completion handlers
+        which do not have an explicit associated executor.
+    */
+    typedef typename AsyncWriteStream::executor_type executor_type;
+
+    /** Get the executor associated with the object.
```

Comment:
The brief for functions which return a meaningful value should usually start with the word "Return"